server/cluster/cluster.go

```diff
@@ -1114,6 +1114,37 @@ func (c *RaftCluster) buryStore(storeID uint64) error {
 	return err
 }
 
+// ForceRemoveStore set a store as tombstone directly without waiting for the data to be moved somewhere else.
+func (c *RaftCluster) ForceRemoveStore(storeID uint64) error {
+	c.Lock()
+	defer c.Unlock()
+
+	store := c.GetStore(storeID)
+	if store == nil {
+		return errs.ErrStoreNotFound.FastGenByArgs(storeID)
+	}
+
+	// Bury a tombstone store should be OK, nothing to do.
+	if store.IsTombstone() {
```

Comment:
and check if the store is up now, if that, reject to remove the store

Reply:
I intentionally created a new function (in addition to `buryStore`) which does not check if the store is up or not. Because, it seems that, whether or not a store is [up or offline](https://github.com/tikv/pd/blob/3448e55fc5e7974e261a282f66204379e70a7855/server/core/store.go#L150) is totally set by the user/admin, while the background liveness check sets [a different state here](https://github.com/tikv/pd/blob/3448e55fc5e7974e261a282f66204379e70a7855/server/core/store.go#L475). If we really want to check the liveness of a store, we probably should check `IsDisconnected()`. Updated, PTAL

Reply:
I intentionally created a new function (in addition to `buryStore`) which does not check if the store is up or not. Because, it seems that, whether or not a store is [up or offline](https://github.com/tikv/pd/blob/3448e55fc5e7974e261a282f66204379e70a7855/server/core/store.go#L150) is totally set by the user/admin, while the background liveness check sets [a different state here](https://github.com/tikv/pd/blob/3448e55fc5e7974e261a282f66204379e70a7855/server/core/store.go#L475). If we really want to check the liveness of a store, we probably should check `IsDisconnected()`. Updated, PTAL